### PR TITLE
Document update

### PR DESCRIPTION
**Pull request by Chibikuri**
# Updated Documents
Updated documents of 

- BSA nodes
- Connection Manager
- Repeater

---
This change is [Reviewable](https://reviewable.io/reviews/sfc-aqua/quisp/309)

### CODE REVIEW

File: doc/Repeater.md [hunk 1]
```diff
@@ -1,8 +1,63 @@
+# Quantum Repeater
 
-*Note: this file is an early design document, but might be updated as the version goes up.*
+## What is this?
 
-# Repeater #
+Quantum Repeater is one crucial network device in entanglement-based quantum networking, which allows us to extend link-level entanglement to end-to-end entanglement.
```
Comment (rdviii) [inline doc/Repeater.md:5]: Based on our network architecture paper, a repeater has exactly two interfaces. I think the QuISP code currently makes no such distinction?

File: doc/Repeater.md [hunk 2]
```diff
@@ -1,8 +1,63 @@
+# Quantum Repeater
 
-*Note: this file is an early design document, but might be updated as the version goes up.*
+## What is this?
 
-# Repeater #
+Quantum Repeater is one crucial network device in entanglement-based quantum networking, which allows us to extend link-level entanglement to end-to-end entanglement. 
 
-A Repeater node has exactly two external links, so that it is useful in a line only.  It is essentially a repurposed [QNIC](QNIC), with both halves of the card running the same [Network Type](Network Type). It also contains a simple [C4](C4).
+The main works of quantum repeaters are following.
```
Comment (rdviii) [inline doc/Repeater.md:7]: Here, in fact, the lines describing the Repeater implementation as having two interfaces has been deleted!

File: doc/Repeater.md [hunk 3]
```diff
@@ -1,8 +1,63 @@
...
+
+At the end of this process, the repeater must tell the result of Bell measurement to proper partners to correct state with Pauli operations. 
+
+### 3. Error management
+
+Quantum error is one critical obstacle for large-scale quantum networking. Keeping the fidelity of the quantum state is also one important task of a quantum repeater. For the near future (called 1G quantum networking), entanglement purification plays an important role to find errors in the quantum state and improving its quality.
+
+### 4. Network operations
+
+The repeater node behaves as an ordinal network node that is responsible for routing, multiplexing, etc. As mentioned above, repeaters must be capable of coordinating operations with other repeaters or end nodes.
```
Comment (rdviii) [inline doc/Repeater.md:36]: Here, a Repeater is said to participate in routing, which is only true in a limited way for two-interface Repeaters.

File: doc/BSA Node.md [hunk 4]
```diff
@@ -1,33 +1,72 @@
...
+
+# What is the interface to the outside world?
+
+BSA provides usually two interfaces through fiber or laser to take in two photons. 
+
+# Who interacts with this interface?
+
+There are two cases.
+
+1. BSA is inside one of the nodes. (MM link)
```
Comment (rdviii) [inline doc/BSA Node.md:39]: Here, we are possibly confusing the issue of the BSA *Node* (which is right there in the name of the .md file) and the BSA (or HOM) functionality.

**Review by rdviii — CHANGES_REQUESTED** [review]
Most of this is good, but we probably need to resolve the BSA/HOM, function/node confusion.
This could actually be accepted as-is, with those changes deferred to another pull request combining both code and documentation changes.

File: doc/Repeater.md (hunk 1 above)
Comment (Chibikuri) [reply doc/Repeater.md:5]: Repeater always has two QNICs in QuISP. Is that what you mean here?

File: doc/Repeater.md (hunk 2 above)
Comment (Chibikuri) [reply doc/Repeater.md:7]: OK, let me add it again.

File: doc/Repeater.md (hunk 3 above)
Comment (Chibikuri) [reply doc/Repeater.md:36]: OK, I'll fix it.

**Review by Naphann — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sfc-aqua/quisp/309)* status: 0 of 12 files reviewed, 8 unresolved discussions (waiting on @Chibikuri, @Naphann, and @rdviii)

___
*[doc/BSA Node.md, line 23 at r2](https://reviewable.io/reviews/sfc-aqua/quisp/309#-MtvKMQS8v9qauhlOUMQ:-MtvKMQS8v9qauhlOUMR:brmhfga) ([raw file](https://github.com/sfc-aqua/quisp/blob/e72b510d989df6abd0f306d0c1c8e4f778573a37/doc/BSA%20Node.md#L23)):*
> ```Markdown
> 
> 
> Here, we assume those entangled states are  known as one of the Bell states, where  is horizontal polarization state and  is vertical polarization state. 
> ```

The rendered latex would look better if you add `\left` and `\right`. 

\left|\psi^{-}\right\rangle=\frac{\left|HV\right\rangle-\left|VH\right\rangle}{\sqrt{2}} Also same thing for all other places. 



_Code quote:_
```Markdown
https://render.githubusercontent.com/render/math?math=|\psi^{-}\rangle = \frac{|HV\rangle - |VH\rangle}{\sqrt{2}}
```
___
*[doc/BSA Node.md, line 37 at r2](https://reviewable.io/reviews/sfc-aqua/quisp/309#-MtvPJ104aZw8V-dwh8a:-MtvPJ104aZw8V-dwh8b:b-dggw9z) ([raw file](https://github.com/sfc-aqua/quisp/blob/e72b510d989df6abd0f306d0c1c8e4f778573a37/doc/BSA%20Node.md#L37)):*
> ```Markdown
> # Who interacts with this interface?
> 
> There are two cases.
> ```

I think MSM also needs BSAs, two of them actually! They are both internal to entangled the incoming photon from source with its own memory (matter) qubits.
___
*[doc/BSA Node.md, line 62 at r2](https://reviewable.io/reviews/sfc-aqua/quisp/309#-MtvNiaW-sXF6KBCnILj:-MtvNiaW-sXF6KBCnILk:b-c9gl7v) ([raw file](https://github.com/sfc-aqua/quisp/blob/e72b510d989df6abd0f306d0c1c8e4f778573a37/doc/BSA%20Node.md#L62)):*
> ```Markdown
> [2] Osawa, Shuto, David S. Simon, and Alexander V. Sergienko. "Directionally-unbiased unitary optical devices in discrete-time quantum walks." Entropy 21.9 (2019): 853.
> 
> [3][https://copilot.caltech.edu/documents/16791/weihs_zeillinger_photon_statistics_at_beamsplitters_qip.pdf](https://copilot.caltech.edu/documents/16791/weihs_zeillinger_photon_statistics_at_beamsplitters_qip.pdf)
> ```

just for consistency, a space after `[3]` and `[4]` 
___
*[doc/Connection Manager.md, line 27 at r2](https://reviewable.io/reviews/sfc-aqua/quisp/309#-MtvOEi26Hq3oiDMHgD9:-MtvOEi26Hq3oiDMHgDA:b3d84wl) ([raw file](https://github.com/sfc-aqua/quisp/blob/e72b510d989df6abd0f306d0c1c8e4f778573a37/doc/Connection%20Manager.md#L27)):*
> ```Markdown
> ### Connection Setup Rejection
> 
> When the link or node is somehow unable to use for the process (occupied by other link, connection unreachable, etc.), node has to reject the request from the initiator. Then, the request must be issued again to check if the link became available or not. 
> ```

"occupied by other link" - this would get rejected if we only talk about circuit switch network. We should note it somewhere to change this when we have multiplexing scheme that works.

File: doc/Repeater.md (hunk 1 above)
Comment (rdviii) [reply doc/Repeater.md:5]: Yes.

Comment (rdviii) [reply doc/Repeater.md:5]: Although one question is, if internally the Repeater has two QNICs, how do we create entanglement between the QNICs? That's not really needed in a lot of Repeater implementations, though it is Router implementations that use an optical switch and multiple independent line cards.
In a Router, each QNIC has two optical connections -- one to the wide-area fiber, one to the optical backplane. So a Repeater is effectively a standalone linecard.

**Pull request closed without merging**